pkg/types/querybuildertypes/querybuildertypesv5/functions.go

```diff
@@ -91,25 +91,77 @@ func ApplyFunction(fn Function, result *TimeSeries) *TimeSeries {
 		// the anomaly alert
 		return result
 	case FunctionNameFillZero:
-		// fillZero expects 3 arguments: start, end, step (all in milliseconds)
-		if len(args) < 3 {
+		start, end, step, ok := resolveFillZeroParams(args, result)
+		if !ok {
 			return result
 		}
-		start, err := parseFloat64Arg(args[0].Value)
-		if err != nil {
-			return result
+		return funcFillZero(result, start, end, step)
+	}
+	return result
+}
+
+// resolveFillZeroParams extracts start, end, and step for fillZero.
+// If start or end are missing, it falls back to the series' min and max timestamps.
+func resolveFillZeroParams(args []FunctionArg, series *TimeSeries) (int64, int64, int64, bool) {
```

Comment:
This should never happen because every query should have the start and end time stamp (which then means steps will also be created), otherwise it's not possible to run the query.